src/provable/from-layout.ts

```diff
@@ -123,19 +120,30 @@ function ProvableFromLayout<
       return json.map((json: any) => fromJSON(arrayTypeData.inner, json));
     }
     if (typeData.type === 'option') {
-      let { optionType, inner } = typeData as OptionLayout<TypeMap>;
-      switch (optionType) {
+      let optionTypeData = typeData as OptionLayout<TypeMap>;
+      switch (optionTypeData.optionType) {
+        case 'closedInterval':
         case 'flaggedOption': {
-          if (json === null) {
-            return { isSome: false, value: emptyValue(inner) };
+          let isSome = TypeMap.Bool.fromJSON(json !== null);
+          let value;
+          if (json !== null) {
+            value = fromJSON(optionTypeData.inner, json);
+          } else {
+            value = emptyValue(optionTypeData.inner);
```

Comment:
these changes fix `fromJSON` in the case the JSON has a `null` value in place of a precondition that is specified as a closed interval. the new implementation recreates the proper default interval which is bounded by the min and max values. these values are annotated in OCaml and read off the layout here